### ISSUE TITLE
[vhost/vhost_initiator] vhost fuzzer fuzz_vhost_create_dev  error

### ISSUE THREAD

**Issue by karlatec** [spdk/spdk#1846]
## Current Behavior

Vhost fuzz test fails at `fuzz_vhost_create_dev` case.

## Steps to Reproduce

1. Run `./test/vhost/fuzz/fuzz.sh` test.

Core dump:
```
gdb: warning: Couldn't determine a path for the index cache directory.
[New LWP 22028]
[New LWP 22031]
[New LWP 22030]
[New LWP 22042]
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".
Core was generated by `/home/vagrant/spdk_repo/spdk/build/bin/vhost -p 0'.
Program terminated with signal SIGABRT, Aborted.
#0  0x00007f40d09bb9d5 in raise () from /lib64/libc.so.6
[Current thread is 1 (Thread 0x7f40d03bd9c0 (LWP 22028))]

Thread 4 (Thread 0x7f40cebb2640 (LWP 22042)):
#0  0x00007f40d0a74a5f in poll () from /lib64/libc.so.6
No symbol table info available.
#1  0x00007f40d155aebf in fdset_event_dispatch (arg=0x7f40d16332e0 <vhost_user+8192>) at ../lib/librte_vhost/fd_man.c:252
        i = 7
        pfd = 0x7f40d1633310 <vhost_user+8240>
        pfdentry = 0x7f40d16353d0 <vhost_user+16624>
        rcb = 0x7f40d156bb06 <vhost_user_read_cb>
        wcb = 0x0
        dat = 0x7f40b800b810
        fd = 315
        numfds = 7
        remove1 = 0
        remove2 = 0
        need_shrink = 0
        pfdset = 0x7f40d16332e0 <vhost_user+8192>
        val = 0
#2  0x00007f40d17d4822 in ctrl_thread_init (arg=0xe82f60) at ../lib/librte_eal/common/eal_common_thread.c:193
        ret = -1
        internal_conf = 0x7f40d182d4c0 <internal_config>
        cpuset = 0x7f40d18305f8 <internal_config+12600>
        params = 0xe82f60
        start_routine = 0x7f40d155ae3e <fdset_event_dispatch>
        routine_arg = 0x7f40d16332e0 <vhost_user+8192>
#3  0x00007f40d0b523f9 in start_thread () from /lib64/libpthread.so.0
No symbol table info available.
#4  0x00007f40d0a7fb53 in clone () from /lib64/libc.so.6
No symbol table info available.

Thread 3 (Thread 0x7f40d03b9640 (LWP 22030)):
#0  0x00007f40d0a7fe9e in epoll_wait () from /lib64/libc.so.6
No symbol table info available.
#1  0x00007f40d17ee162 in eal_intr_handle_interrupts (pfd=8, totalfds=1) at ../lib/librte_eal/linux/eal_interrupts.c:1045
        events = {{events = 0, data = {ptr = 0x0, fd = 0, u32 = 0, u64 = 0}}}
        nfds = 0
#2  0x00007f40d17ee35c in eal_intr_thread_main (arg=0x0) at ../lib/librte_eal/linux/eal_interrupts.c:1130
        pipe_event = {events = 3, data = {ptr = 0x4, fd = 4, u32 = 4, u64 = 4}}
        src = 0x0
        numfds = 1
        pfd = 8
        __func__ = "eal_intr_thread_main"
#3  0x00007f40d17d4822 in ctrl_thread_init (arg=0xdbb020) at ../lib/librte_eal/common/eal_common_thread.c:193
        ret = -1
        internal_conf = 0x7f40d182d4c0 <internal_config>
        cpuset = 0x7f40d18305f8 <internal_config+12600>
        params = 0xdbb020
        start_routine = 0x7f40d17ee1c4 <eal_intr_thread_main>
        routine_arg = 0x0
#4  0x00007f40d0b523f9 in start_thread () from /lib64/libpthread.so.0
No symbol table info available.
#5  0x00007f40d0a7fb53 in clone () from /lib64/libc.so.6
No symbol table info available.

Thread 2 (Thread 0x7f40cfbb8640 (LWP 22031)):
#0  0x00007f40d0b5c22f in accept () from /lib64/libpthread.so.0
No symbol table info available.
#1  0x00007f40d164c804 in socket_listener (socket=0x7f40d16517e0 <v2_socket>) at ../lib/librte_telemetry/telemetry.c:345
        th = 0
        s = 0x7f40d16517e0 <v2_socket>
        s_accepted = 0
#2  0x00007f40d0b523f9 in start_thread () from /lib64/libpthread.so.0
No symbol table info available.
#3  0x00007f40d0a7fb53 in clone () from /lib64/libc.so.6
No symbol table info available.

Thread 1 (Thread 0x7f40d03bd9c0 (LWP 22028)):
#0  0x00007f40d09bb9d5 in raise () from /lib64/libc.so.6
No symbol table info available.
#1  0x00007f40d09a48a4 in abort () from /lib64/libc.so.6
No symbol table info available.
#2  0x00007f40d09a4789 in __assert_fail_base.cold () from /lib64/libc.so.6
No symbol table info available.
#3  0x00007f40d09b4026 in __assert_fail () from /lib64/libc.so.6
No symbol table info available.
#4  0x00007f40d1d05fba in scsi2_release (task=0x200023cb6f90) at scsi_pr.c:1030
        lun = 0xe84890
        ret = 0
        __PRETTY_FUNCTION__ = "scsi2_release"
#5  0x00007f40d1cfd7ef in bdev_scsi_process_primary (task=0x200023cb6f90) at scsi_bdev.c:1946
        lun = 0xe84890
        bdev = 0xe83950
        alloc_len = -1
        data_len = -1
        cdb = 0x7f40b7ae13f3 "W\262\225\230\004O\250\344\314\337q\244\326\310p\372B\257\341!\234\214\212m\247=Jx\256\032vP"
        data = 0x0
        rc = 0
        pllen = 950863733
        md = 0
        llba = 0
        dbd = -1213328397
        pc = 32576
        page = 1413481970
        subpage = 32767
        cmd_parsed = 0
        __PRETTY_FUNCTION__ = "bdev_scsi_process_primary"
        __func__ = "bdev_scsi_process_primary"
#6  0x00007f40d1cfd9aa in bdev_scsi_execute (task=0x200023cb6f90) at scsi_bdev.c:1977
        rc = -1
        __func__ = "bdev_scsi_execute"
#7  0x00007f40d1cec5e3 in _scsi_lun_execute_task (lun=0xe84890, task=0x200023cb6f90) at lun.c:209
        rc = 0
#8  0x00007f40d1cece7e in scsi_lun_execute_task (lun=0xe84890, task=0x200023cb6f90) at lun.c:263
No locals.
#9  0x00007f40d1ce91cf in spdk_scsi_dev_queue_task (dev=0x7f40d1d249e0 <g_devs>, task=0x200023cb6f90) at dev.c:292
        __PRETTY_FUNCTION__ = "spdk_scsi_dev_queue_task"
#10 0x00007f40d204cc5c in task_submit (task=0x200023cb6f90) at vhost_scsi.c:381
No locals.
#11 0x00007f40d205131e in process_scsi_task (vsession=0x7f40b800b840, vq=0x7f40b800b9c0, req_idx=451) at vhost_scsi.c:724
        task = 0x200023cb6f90
        result = 0
        __func__ = "process_scsi_task"
#12 0x00007f40d2052490 in process_vq (svsession=0x7f40b800b840, vq=0x7f40b800b9c0) at vhost_scsi.c:799
        vsession = 0x7f40b800b840
        reqs = {448, 451, 454, 457, 460, 463, 466, 469, 472, 475, 478, 481, 484, 487, 490, 493, 496, 499, 502, 505, 508, 511, 514, 517, 520, 523, 526, 529, 532, 535, 538, 541}
        reqs_cnt = 32
        i = 1
        __PRETTY_FUNCTION__ = "process_vq"
        __func__ = "process_vq"
#13 0x00007f40d20528aa in vdev_worker (arg=0x7f40b800b840) at vhost_scsi.c:831
        svsession = 0x7f40b800b840
        vsession = 0x7f40b800b840
        q_idx = 2
#14 0x00007f40d1a50139 in thread_poll (thread=0xe84400, max_msgs=0, now=643389890511) at thread.c:653
        poller_rc = 1
        msg_count = 0
        poller = 0xe7ea80
        tmp = 0x0
        critical_msg = 0x0
        rc = 0
        __func__ = "thread_poll"
#15 0x00007f40d1a51e49 in spdk_thread_poll (thread=0xe84400, max_msgs=0, now=643389890511) at thread.c:741
        orig_thread = 0x0
        rc = 1
#16 0x00007f40d1a9d48f in _reactor_run (reactor=0xe784c0) at reactor.c:888
        thread = 0xe84400
        lw_thread = 0xe84740
        tmp = 0xe85bd0
        now = 643389890511
        rc = 1
#17 0x00007f40d1a9da95 in reactor_run (arg=0xe784c0) at reactor.c:926
        reactor = 0xe784c0
        thread = 0x7f40d17c4b0a <rte_get_next_lcore+42>
        lw_thread = 0x7f00000000
        tmp = 0x7fff54400910
        thread_name = "reactor_0\000\000\000\200\000\000\000@\t@T\377\177\000\000\220\000\221\321\000\000\000"
        last_sched = 0
        __func__ = "reactor_run"
#18 0x00007f40d1a9e57a in spdk_reactors_start () at reactor.c:1034
        reactor = 0xe784c0
        i = 4294967295
        current_core = 0
        rc = 0
        __func__ = "spdk_reactors_start"
        __PRETTY_FUNCTION__ = "spdk_reactors_start"
#19 0x00007f40d1a94d87 in spdk_app_start (opts_user=0x7fff54400c60, start_fn=0x40255e <vhost_started>, arg1=0x0) at app.c:577
        rc = 0
        tty = 0x0
        tmp_cpumask = {str = '\000' <repeats 256 times>, cpus = "\001", '\000' <repeats 126 times>}
        g_env_was_setup = false
        opts_local = {name = 0x4050d7 "vhost", json_config_file = 0x0, json_config_ignore_errors = false, rpc_addr = 0x7f40d1ab41c0 "/var/tmp/spdk.sock", reactor_mask = 0x7f40d1ab41bc "0x1", tpoint_group_mask = 0x0, shm_id = -1, shutdown_cb = 0x0, enable_coredump = true, mem_channel = -1, main_core = 0, mem_size = -1, no_pci = false, hugepage_single_segments = false, unlink_hugepage = false, hugedir = 0x0, print_level = SPDK_LOG_INFO, num_pci_addr = 0, pci_blocked = 0x0, pci_allowed = 0x0, iova_mode = 0x0, delay_subsystem_init = false, num_entries = 32768, env_context = 0x0, log = 0x0, base_virtaddr = 35184372088832, opts_size = 184}
        opts = 0x7fff544009d0
        __func__ = "spdk_app_start"
#20 0x00000000004026cb in main (argc=3, argv=0x7fff54400e18) at vhost.c:105
        opts = {name = 0x4050d7 "vhost", json_config_file = 0x0, json_config_ignore_errors = false, rpc_addr = 0x7f40d1ab41c0 "/var/tmp/spdk.sock", reactor_mask = 0x7f40d1ab41bc "0x1", tpoint_group_mask = 0x0, shm_id = -1, shutdown_cb = 0x0, enable_coredump = true, mem_channel = -1, main_core = 0, mem_size = -1, no_pci = false, hugepage_single_segments = false, unlink_hugepage = false, hugedir = 0x0, print_level = SPDK_LOG_INFO, num_pci_addr = 0, pci_blocked = 0x0, pci_allowed = 0x0, iova_mode = 0x0, delay_subsystem_init = false, num_entries = 32768, env_context = 0x0, log = 0x0, base_virtaddr = 35184372088832, opts_size = 184}
        rc = 1
```

Files:
[logfiles.zip](https://github.com/spdk/spdk/files/6181390/logfiles.zip)
[coredumps.zip](https://github.com/spdk/spdk/files/6181387/coredumps.zip)

## Context (Environment including OS version, SPDK version, etc.)
SPDK `b8db1af4e2` `thread: criticalmsg also indicates valid event`
Fedora 33 5.10.23-200.fc33.x86_64
gcc version 10.2.1 20201125 (Red Hat 10.2.1-9) (GCC)

**Comment by tomzawadzki** [comment]
Bug scrub: This started appearing after updating from Fedora 31 to Fedora 33, does not have 100% reproduction.

**Comment by ksztyber** [comment]
Already fixed by 44c70f8215cc87fa55a949df2e771ff2cb2da1ed.
